q.js

```diff
@@ -1752,6 +1752,9 @@ Q["finally"] = function (object, callback) {
 
 Promise.prototype.fin = // XXX legacy
 Promise.prototype["finally"] = function (callback) {
+    if (!callback || typeof callback.apply !== "function") {
```

Comment:
I find this a little confusing, but I feel the alternative (inverting the conditional, and having the guard at the bottom) is worse